### PR TITLE
chore/interface-estlint

### PR DESCRIPTION
**Pull request by PaulieScanlon**
- update eslint rules

### CODE REVIEW

File: packages/mdx-embed/.eslintrc.js [hunk 1]
```diff
@@ -17,12 +17,23 @@ module.exports = {
...
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/no-empty-interface': 'warn',
+    '@typescript-eslint/no-non-null-assertion': 'off',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: {
+          regex: '^I[A-Z]',
```
Comment (PaulieScanlon) [inline packages/mdx-embed/.eslintrc.js:33]: What i really want is to lint the TypeScript interface names so they are always `ISomethingProps` but i can't seem to get this regex to work: `^I[A-Z]\w+Props\b`

Any ideas?

Comment (daisygabi) [reply packages/mdx-embed/.eslintrc.js:33]: Hi, I just checked the regex online and it looks like it's correct. Using https://regexr.com/ I tried with "iTechProps" and "ITechprops" and it wasn't a match -> correct as well.
rege

Comment (PaulieScanlon) [reply packages/mdx-embed/.eslintrc.js:33]: @daisygabi Thanks, i was using [https://regex101.com/](https://regex101.com/) and it was this worked `^I[A-Z]\w+Props\b` i wonder if it's prettier doing something weird

Comment (drydenwilliams) [reply packages/mdx-embed/.eslintrc.js:33]: Here ya go: `I([A-Z][a-zA-Z]+|)Props`

Screenshot 2020-09-16 at 10 22 28

Comment (PaulieScanlon) [reply packages/mdx-embed/.eslintrc.js]: Boogy time Thank you @drydenwilliams ... just for completeness: This is covers `State` too. e.g `IMyTestProps` | `IMyTestState`

```
I(([A-Z][a-z]+)+|)(Props|State)
```

**Pull request merged**